lib/Collection/QueryType/Handler/TaxonsHandler.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Layouts\Sylius\Collection\QueryType\Handler;
+
+use Netgen\Layouts\API\Values\Collection\Query;
+use Netgen\Layouts\Collection\QueryType\QueryTypeHandlerInterface;
+use Netgen\Layouts\Parameters\ParameterBuilderInterface;
+use Netgen\Layouts\Parameters\ParameterType;
+use Netgen\Layouts\Sylius\Parameters\ParameterType as SyliusParameterType;
+use Sylius\Component\Taxonomy\Model\TaxonInterface;
+use Netgen\Layouts\Sylius\Doctrine\ORM\TaxonRepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+use function trim;
+
+final class TaxonsHandler implements QueryTypeHandlerInterface
+{
+    public function __construct(
+        private TaxonRepositoryInterface $taxonRepository,
+        private RequestStack $requestStack,
+    ) {}
+
+    public function buildParameters(ParameterBuilderInterface $builder): void
+    {
+        $builder->add(
+            'use_current_taxon',
+            ParameterType\Compound\BooleanType::class,
+            [
+                'reverse' => true,
+            ],
+        );
+
+        $builder->get('use_current_taxon')->add(
+            'taxon_id',
+            SyliusParameterType\TaxonType::class,
+        );
+    }
+
+    public function getValues(Query $query, int $offset = 0, ?int $limit = null): iterable
+    {
+        $currentRequest = $this->requestStack->getCurrentRequest();
+        if (!$currentRequest instanceof Request) {
+            return [];
+        }
+
+        $taxon = $this->getTaxon($query);
+
+        if (!$taxon instanceof TaxonInterface) {
+            return [];
+        }
+
+        $taxonCode = $taxon->getCode();
+        if ($taxonCode === null) {
+            return [];
+        }
+
+        /** @var \Sylius\Component\Taxonomy\Model\TaxonInterface[] $children */
```

Comment:
This one is tricky, with current code the cs fixer is complaining that it could be a one liner:
return $this->taxonRepository->findChildren($taxonCode);

however - findChildren method in the vendor returns array|TaxonInterface[] without specifying what is returned in array after which the psalm and phpstan are complaining

Reply:
Okay, we will leave it as is.